### PR TITLE
Proxy switch for sensu-install

### PR DESCRIPTION
**Pull request by seifer44**
Added in support to utilize the proxy option for `gem`.

## Description
The sensu-install utility was unable to install the sensu plugins via a proxy server. This rectifies that issue.

## Related Issue
https://github.com/sensu/sensu/issues/1588

## Motivation and Context
Allows the use of proxy servers for sensu-install.

## How Has This Been Tested?
This was tested against two RHEL 7 servers.

1. A server behind a proxy, without direct access to the internet. Installations failed before, they work with the proposed changes.
2. A server with direct access to the internet, without providing any options to the proxy variable. This worked as expected.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x ] My change requires a change to the documentation.
- [ x ] I have updated the documentation accordingly.
- [ x ] I have read the **CONTRIBUTING** document.
- [ x ] I have added tests to cover my changes.
- [ x ] All new and existing tests passed.

### CODE REVIEW

File: exe/sensu-install [hunk 1]
```diff
@@ -70,6 +74,7 @@ module Sensu
         gem_command << " --no-ri --no-rdoc"
         gem_command << " --verbose" if options[:verbose]
         gem_command << " --source #{options[:source]}" if options[:source]
+        gem_command << " --http-proxy #{options[:proxy]}"
```
Comment (portertech) [inline exe/sensu-install]: needs an `if options[:proxy]` conditional after making the other changes.

File: exe/sensu-install [hunk 2]
```diff
@@ -10,7 +10,8 @@ module Sensu
           :verbose => false,
           :plugins => [],
           :extensions => [],
-          :clean => false
+          :clean => false,
+          :proxy => ""
```
Comment (portertech) [inline exe/sensu-install]: default to `nil`?

File: exe/sensu-install [hunk 3]
```diff
@@ -38,6 +39,9 @@ module Sensu
           opts.on("-c", "--clean", "Clean up (remove) other installed versions of the plugin(s) and/or extension(s)") do
             options[:clean] = true
           end
+          opts.on("-x", "--proxy PROXY", "Install plugins via a PROXY url") do |proxy|
+            options[:proxy] << proxy
```
Comment (portertech) [inline exe/sensu-install]: change `<<` to `=`?

**Comment by seifer44** [comment]
Those changes have been made. Thanks for the recommendations; I'm not familiar with Ruby at all, if that wasn't apparent. I was just hacking away and wanted it to work for me :)

**Comment by portertech** [comment]
94bf04ce-44e5-11e4-99ae-7fa5af8c9e8b

**Pull request merged**